lib/asciidoctor-epub3/packager.rb

```diff
@@ -615,24 +615,35 @@ def package options = {}
         end
       end
 
-      def distill_epub_to_mobi epub_file, target, compress
-        if !(kindlegen_cmd = ENV['KINDLEGEN']).nil?
-          argv = [kindlegen_cmd]
-        else
-          begin
-            require 'kindlegen' unless defined? ::Kindlegen
-          rescue LoadError => e
-            raise 'Unable to find KindleGen. Either install the kindlegen gem or set KINDLEGEN environment variable with path to KindleGen executable', cause: e
-          end
-          argv = [::Kindlegen.command.to_s]
+      def get_kindlegen_command
+        unless (result = ENV['KINDLEGEN']).nil?
```

Comment:
I would like that besides finding it in the environment there is another way of locating the binary i.e. via an attribute. This allows external downloaders to set the path. For instance in Gradle we are forced to run this out of process i.e. another JVM because of the environmental variable, but if there is a way of setting it, then we can utilise in-process generation.

Reply:
You mean, via document attribute?

Reply:
@mojavelinux could suggest attribute name for this so we don't acidentally clash with other asciidoctor machinery?

Reply:
@ysb33r I think, the order will be:

1. env var
2. attribute
3. kindlegen gem
4. PATH

Reply:
After more thinking, I want to make attribute higher priority than env var. Case: user has env var defined and calls `asciidoctor-epub3 -a ebook-kindlegen-binary-path=/path <book>`. It is clear that their intent to override global behavior for this particular invocation.

Reply:
@mojavelinux are there any security concerns about passing kindlegen path through document attributes? Is there any alternative option to pass settings to `Asciidoctor::Epub3::Packager` that bypass the document itself?

Reply:
@ysb33r I've updated PR with document attributes to specify KindleGen/EPUBCheck executable location, but still waiting for @mojavelinux to review this.